lib/src/execution/windows/execution.cc

```diff
@@ -497,21 +513,28 @@ namespace facter { namespace execution {
 
         // Use a Job Object to group any child processes spawned by the CreateProcess invocation, so we can
         // easily stop them in case of a timeout.
-        scoped_resource<HANDLE> hJob(CreateJobObjectW(nullptr, nullptr), CloseHandle);
-        if (hJob == NULL) {
-            LOG_ERROR("failed to create job object: %1%.", system_error());
-            throw execution_exception("failed to create job object.");
-        } else if (!AssignProcessToJobObject(hJob, hProcess)) {
-            LOG_ERROR("failed to associate process with job object: %1%.", system_error());
-            throw execution_exception("failed to associate process with job object.");
+        scoped_resource<HANDLE> hJob;
+        if (use_job_object) {
+            hJob = scoped_resource<HANDLE>(CreateJobObjectW(nullptr, nullptr), CloseHandle);
+            if (hJob == NULL) {
+                LOG_ERROR("failed to create job object: %1%.", system_error());
+                throw execution_exception("failed to create job object.");
+            } else if (!AssignProcessToJobObject(hJob, hProcess)) {
+                LOG_ERROR("failed to associate process with job object: %1%.", system_error());
+                throw execution_exception("failed to associate process with job object.");
+            }
         }
 
         bool terminate = true;
         scope_exit reaper([&]() {
             if (terminate) {
                 // Terminate the process on an exception
-                if (!TerminateJobObject(hJob, -1)) {
-                    LOG_ERROR("failed to terminate process: %1%.", system_error());
+                if (use_job_object) {
+                    if (!TerminateJobObject(hJob, -1)) {
+                        LOG_ERROR("failed to terminate process: %1%.", system_error());
+                    }
+                } else {
+                    LOG_WARNING("could not terminate process: %1% because a job object could not be used.", procInfo.dwProcessId);
```

Comment:
Nit pick: this reads a bit strange.  Perhaps `could not terminate process %1%: a job object could not be used.`?